src/main/java/no/sikt/nva/BrageProcessorFactory.java

```diff
@@ -1,21 +1,24 @@
 package no.sikt.nva;
 
+import java.util.Map;
 import nva.commons.core.StringUtils;
 
-public final class BrageProcessorFactory {
+public class BrageProcessorFactory {
 
     public static final String ZIP_EXTENSION = ".zip";
     private static final String INVALID_ZIPFILE_NAME_EXCEPTION_MESSAGE = "invalid zipfile name";
 
-    private BrageProcessorFactory() {
+    private final Map<String, String> rescueTitleAndHandleMap;
 
+    public BrageProcessorFactory(Map<String, String> rescueTitleAndHandleMap) {
+        this.rescueTitleAndHandleMap = rescueTitleAndHandleMap;
     }
 
-    public static BrageProcessor createBrageProcessor(String zipfile, String customerId) {
+    public BrageProcessor createBrageProcessor(String zipfile, String customerId) {
         var destinationDirectory = zipfile.replace(ZIP_EXTENSION, StringUtils.EMPTY_STRING);
         if (StringUtils.isEmpty(destinationDirectory)) {
             throw new RuntimeException(INVALID_ZIPFILE_NAME_EXCEPTION_MESSAGE);
         }
-        return new BrageProcessor(zipfile, customerId, destinationDirectory);
+        return new BrageProcessor(zipfile, customerId, destinationDirectory, rescueTitleAndHandleMap);
```

Comment:
*rescue litt komplisert ord for naming av funksjon? kanskje det er bare meg som ikke kan engelsk